graphql_compiler/query_formatting/common.py

```diff
@@ -28,6 +28,63 @@ def _raise_invalid_type_error(name, expected_python_type_name, value):
                                                          type(value).__name__))
 
 
+def _deserialize_anonymous_json_argument(expected_type, value):
+    """Deserialize argument. See docstring of deserialize_json_argument."""
+    allowed_types_for_graphql_type = {
+        GraphQLDate: (six.string_types,),
+        GraphQLDateTime: (six.string_types,),
+        GraphQLFloat: (six.string_types, float, six.integer_types),
+        GraphQLDecimal: (six.string_types, float, six.integer_types),
+        GraphQLInt: (six.integer_types,),
+        GraphQLString: (six.string_types,),
+        GraphQLBoolean: (bool,),
+        GraphQLID: (six.string_types,),
+    }
+
+    # Check if the type of the value is correct
+    correct_type = True
+    expected_python_types = allowed_types_for_graphql_type[expected_type]
+    if isinstance(value, bool) and not GraphQLBoolean.is_same_type(expected_type):
+        correct_type = False  # We explicitly disallow passing boolean values for non-boolean types
+    if not isinstance(value, expected_python_types):
+        correct_type = False
+    if not correct_type:
+        raise ValueError(u'Unexpected type {}. Expected one of {}.'
+                         .format(type(value), expected_python_types))
+
+    # Use the default GraphQL parser to parse the value
+    return expected_type.parse_value(value)
+
+
+def deserialize_json_argument(name, expected_type, value):
+    """Deserialize a GraphQL argument parsed from a json file.
+
+    Passing arguments via jsonrpc, or via the GUI of standard GraphQL editors is tricky because
+    json does not support certain types like Date, Datetime, Decimal, and also confuses floats
+    for integers if there are no decimals. This function takes in a value received from a json,
+    and converts it to a standard python representation that can be used in the query. Below are
+    examples of accepted json encodings of all the types:
+        GraphQLDate: "2018-02-01"
+        GraphQLDateTime: "2018-02-01T05:11:54Z"
```

Comment:
Let's add an example and test case to prove that explicit timezone info is parsed correctly as well: e.g., `2018-02-01T05:11:54Z` and `2018-02-01T05:11:54+00:00` are the same thing, per the standard.